various-receipes/nestjs/app/test/basic-tests.test.ts

```diff
@@ -0,0 +1,61 @@
+import { INestApplication } from '@nestjs/common';
+import { Test } from '@nestjs/testing';
+import { AppModule } from '../app.module';
+const request = require('supertest');
+const sinon = require('sinon');
+const nock = require('nock');
+
+let app: INestApplication;
+
+beforeAll(async (done) => {
+  // ️️️✅ Best Practice: Place the backend under test within the same process
+  const moduleFixture = await Test.createTestingModule({
+    imports: [AppModule],
+  }).compile();
+
+  app = moduleFixture.createNestApplication();
+  await app.init();
+
+  // ️️️✅ Best Practice: Ensure that this component is isolated by preventing unknown calls
+  nock.disableNetConnect();
+  nock.enableNetConnect('127.0.0.1');
+
+  done();
+});
+
+beforeEach(() => {
+  nock('http://localhost/user/').get(`/1`).reply(200, {
```

Comment:
Might confuse the reader since this gun that appears in the 1st scene will do nothing in the last scene?